src/core/dbus-execute.c

```diff
@@ -788,6 +825,8 @@ const sd_bus_vtable bus_exec_vtable[] = {
         SD_BUS_PROPERTY("WorkingDirectory", "s", property_get_working_directory, 0, SD_BUS_VTABLE_PROPERTY_CONST),
         SD_BUS_PROPERTY("RootDirectory", "s", NULL, offsetof(ExecContext, root_directory), SD_BUS_VTABLE_PROPERTY_CONST),
         SD_BUS_PROPERTY("RootImage", "s", NULL, offsetof(ExecContext, root_image), SD_BUS_VTABLE_PROPERTY_CONST),
+        SD_BUS_PROPERTY("RootHash", "ay", property_get_root_hash, 0, SD_BUS_VTABLE_PROPERTY_CONST),
```

Comment:
expose this as two separate props please:

1. `RootHash` as `ay`, which contains the root hash if configurd literrary, but returns an empty array when configured as path
2. `RootHashPath` as `s` which contains the path when configured as path and an empy string otherwise.

Reply:
Sure, will do - is it OK to handle duplication by returning an error if both are set? In other words, only allow to set one property if the other is empty

Reply:
well this property is not writable? it's only writable when creating a new transient unit. On those if both properties are set the latter should take effect, to expose similar behaviour as we'd do when parsing unit files.

In memory, it should not be possible to ever have both fields set. Because when one is set the other should always be reset, and vice versa. You could thus add an assert() verifying this here, if you want, as it would be a programming error in our code if both fields are ever set.

Reply:
Ok, thanks - pushed new update, with all the requested changes applied.

```
root@localhost:~# systemd-run -p RootImage=/tmp/img.squashfs -p RootHash=/tmp/img.roothash -p RootVerity=/tmp/img.hashtree sleep 40
Running as unit: run-r0885dc9f1c6142fa90576c6aa6a7037e.service
root@localhost:~# busctl get-property org.freedesktop.systemd1 /org/freedesktop/systemd1/unit/run_2dr0885dc9f1c6142fa90576c6aa6a7037e_2eservice org.freedesktop.systemd1.Service RootHash
ay 0
root@localhost:~# busctl get-property org.freedesktop.systemd1 /org/freedesktop/systemd1/unit/run_2dr0885dc9f1c6142fa90576c6aa6a7037e_2eservice org.freedesktop.systemd1.Service RootHashPath
s "/tmp/img.roothash"
root@localhost:~# systemd-run -p RootImage=/tmp/img.squashfs -p RootHash=`cat /tmp/img.roothash` -p RootVerity=/tmp/img.hashtree sleep 40
Running as unit: run-rb7b4b77b80624de39e459ee1853977f5.service
root@localhost:~# busctl get-property org.freedesktop.systemd1 /org/freedesktop/systemd1/unit/run_2drb7b4b77b80624de39e459ee1853977f5_2eservice org.freedesktop.systemd1.Service RootHashPath
s ""
root@localhost:~# busctl get-property org.freedesktop.systemd1 /org/freedesktop/systemd1/unit/run_2drb7b4b77b80624de39e459ee1853977f5_2eservice org.freedesktop.systemd1.Service RootHash    
ay 32 131 53 169 171 202 116 131 52 202 171 231 74 26 215 36 63 158 43 137 45 197 171 104 183 161 217 129 123 193 26 85 122
```